src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java

```diff
@@ -101,4 +105,35 @@ public DocAssemblyTemplateBody from(CCDCase ccdCase, UserDetails userDetails) {
                 ccdCase.getExpertReportPermissionPartyGivenToDefendant() == YES)
             .build();
     }
+
+    public DocAssemblyTemplateBody generalLetterBody(CCDCase ccdCase) {
+        LocalDate currentDate = LocalDate.now(clock.withZone(UTC_ZONE));
+        String partyName;
+        CCDAddress partyAddress;
+        if (ccdCase.getApplicants().get(0).getValue()
+            .getPartyName().equals(CCDContactPartyType.CLAIMANT)) {
+            partyName = ccdCase.getApplicants().get(0).getValue().getPartyName();
+            partyAddress = ccdCase.getApplicants().get(0)
+                .getValue().getPartyDetail().getPrimaryAddress();
+        } else {
+            partyName = ccdCase.getRespondents().get(0)
+                .getValue().getClaimantProvidedPartyName();
+            partyAddress = getDefendantAddress(ccdCase.getRespondents().get(0).getValue());
+        }
+        return DocAssemblyTemplateBody.builder()
+            .currentDate(currentDate)
+            .referenceNumber(ccdCase.getPreviousServiceCaseReference())
+            .caseWorkerName(ccdCase.getCaseworkerName())
+            .caseName(ccdCase.getCaseName())
+            .partyName(partyName)
+            .partyAddress(partyAddress)
+            .body(ccdCase.getLetterContent())
+            .build();
+    }
+
+    private CCDAddress getDefendantAddress(CCDRespondent respondent) {
+        return Optional.ofNullable(respondent.getPartyDetail()).isPresent()
```

Comment:
No point wrapping in an Optional just to check for presence.
```suggestion
        return respondent.getPartyDetail() != null
```